client/network/src/discovery.rs

```diff
@@ -407,24 +414,6 @@ impl NetworkBehaviour for DiscoveryBehaviour {
 			list.extend(list_to_filter);
 		}
 
-		if !list.is_empty() {
-			trace!(target: "sub-libp2p", "Addresses of {:?}: {:?}", peer_id, list);
-
-		} else {
-			let mut has_entry = false;
-			for k in self.kademlias.values_mut() {
-				if k.kbuckets_entries().any(|p| p == peer_id) {
-					has_entry = true;
-					break
-				}
-			}
-			if has_entry {
-				trace!(target: "sub-libp2p", "Addresses of {:?}: none (peer in k-buckets)", peer_id);
-			} else {
-				trace!(target: "sub-libp2p", "Addresses of {:?}: none (peer not in k-buckets)", peer_id);
-			}
-		}
-
```

Comment:
As far as I can tell libp2p-kad does ensure at all times that each peer in its kbuckets has at least one address. Thereby `has_entry` is never `true` as otherwise the given peer would have been returned by `k.addresses_of_peer(peer_id)` and thus `list.is_empty` would be `false`.

@romanb am I missing something?

Reply:
Yes, if `addresses_of_peer` returns no addresses for a peer, then that peer is not in the routing table, but the code above seems to also filter out private Ip4 addresses, so in principle `list` maybe be empty despite the peer being in a bucket, but I don't know whether that is a case you want to handle specially?

Reply:
`addresses_of_peer` is also called for bootnodes, reserved nodes, nodes discovered via mDNS, and so on. It occasionally happens that `list` is empty, and the code that you removed helps figure out why we then can't connect to a specific node.

Reply:
`list.is_empty() && has_entry` is only `true` when `peer_id` is a peer in the routing table with private IP addresses only, correct?

In the future, when adding peers to the DHT manually (https://github.com/paritytech/substrate/pull/6549), I would suggest ignoring private IP addresses in the first place. With that in mind would this log line still be useful @tomaka?

Reply:
The check for `has_entry` and the follow-up `if` can indeed be removed.

Reply:
:+1: Addressed in https://github.com/paritytech/substrate/pull/6559/commits/d9373b1209ec5a49766630aff5ef1b622ab9f3f5.